src/main/java/com/targus/ui/WSNPrototype.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public void setPaneWidth(int paneWidth) {
+        this.paneWidth = paneWidth;
+    }
+
+    public void setPaneHeight(int paneHeight) {
+        this.paneHeight = paneHeight;
+    }
+
+    public void exportToFile() {
+        try {
+            FileChooser fc = new FileChooser();
+            fc.setInitialDirectory(new File("."));
+            fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
+            File f = fc.showSaveDialog(null);
+
+            if (f == null) {
+                return;
+            }
+
+            String src = f.getAbsolutePath();
+
+            BufferedWriter writer = Files.newBufferedWriter(Paths.get(src));
+            ObjectMapper objectMapper = new ObjectMapper();
+
+            Map<String, Object> problemInfo = new HashMap<>();
+            problemInfo.put(Constants.DIMENSIONS, Arrays.asList(paneWidth, paneHeight));
+
+            List<double[]> targetList = new ArrayList<>();
+            for (Point2D target : targets) {
+                double[] coords = new double[2];
+                coords[0] = target.getX();
+                coords[1] = target.getY();
+                targetList.add(coords);
+            }
+            problemInfo.put(Constants.TARGETS, targetList);
+
+            List<double[]> potentialPositionList = new ArrayList<>();
+            for (Point2D potentialPosition : potentialPositions) {
+                double[] coords = new double[2];
+                coords[0] = potentialPosition.getX();
+                coords[1] = potentialPosition.getY();
+                potentialPositionList.add(coords);
+            }
+            problemInfo.put(Constants.POTENTIAL_POSITIONS, potentialPositionList);
+
+            problemInfo.put(Constants.COMMUNICATION_RADIUS, communicationRangeProperty.get());
+            problemInfo.put(Constants.SENSING_RADIUS, sensingRangeProperty.get());
+            problemInfo.put(Constants.M, mProperty.get());
+            problemInfo.put(Constants.K, kProperty.get());
+            problemInfo.put(Constants.GENERATION_COUNT, generationCountProperty.get());
+            problemInfo.put(Constants.MUTATION_RATE, mutationRateProperty.get());
+
+            writer.write(objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(problemInfo));
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void loadFromFile() {
+        FileChooser fc = new FileChooser();
+        fc.setInitialDirectory(new File("."));
+        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
+        File f = fc.showOpenDialog(null);
+
+        if (f == null) {
+            return;
+        }
+
+        String src = f.getAbsolutePath();
+
+        Point2D dimensions;
+        Point2D[] targetArray;
+        Point2D[] potentialPositionArray;
+
+        try {
+            Reader reader = Files.newBufferedReader(Paths.get(src));
```

Comment:
The same thing mentioned for JSON write can also be done here. We can write a private method that reads the data

Reply:
Adds to the complexity without a significant gain. So I thought it's better to keep it as is.